src/t8_schemes/t8_new_feature/t8_subelements/t8_subelements_quad_cxx.cxx

```diff
@@ -0,0 +1,1325 @@
+    if (type >= pow (2, (P4EST_FACES - 1) - i)) {
+      binary_array[i] = 1;
+      type -= pow (2, (P4EST_FACES - 1) - i);
+    }
+  } /* we now got a binary represenation of the subelement type, bitwise stored in an array */
+
+  /* 2) rearrange the binary representation to be in clockwise order */
+  int binary_array_temp[P4EST_FACES] = {};
+
+  int j;
+
+  for (j = 0; j < P4EST_FACES; j++) { /* copying the binary array */
+    binary_array_temp[j] = binary_array[j];
+  }
+
+  binary_array[0] = binary_array_temp[0]; /* f3 <- f3 */
+  binary_array[1] = binary_array_temp[3]; /* f2 <- f0 */
+  binary_array[2] = binary_array_temp[1]; /* f1 <- f2 */
+  binary_array[3] = binary_array_temp[2]; /* f0 <- f1 */
+
+  /* 3) use the rearranged binary representation, and the sub_id to determine the location of the subelement and store these information in an array */
+  /* 3.1) location[0] -> the face_number, the subelement is adjacent to */
+  /* 3.2) location[1] -> dummy, if the face is split or not */
+  /* 3.3) location[2] -> is the subelement the first or second subelement of the face (always the first, if the face is not split) */
+  T8_ASSERT (pquad_w_sub->subelement_id < pquad_w_sub->num_subelement_ids);
+
+  int sub_id = pquad_w_sub->subelement_id;
+  int sub_face_id = 1; /* initialize the sub_face_id as the second subelement of a face and change that if it is not true */
+  int face_number = 0;
+  int split;
+
+  int k = 0;
+
+  if (sub_id == 0) {
+    sub_face_id = 0; /* in this case, the subelement is the first element of the face */
+  }
+
+  while (sub_id >= 0 && k < P4EST_FACES) {
+    if (binary_array[k] == 1) {
+      sub_id -= 2; /* two subelements adjacent to face k */
+    }
+    else {
+      sub_id -= 1; /* one subelement adjacent to face k */
+    }
+    if (sub_id == 0) {
+      sub_face_id = 0; /* in this case, the subelement is the first element of the face */
+    }
+    k++;
+  }
+  face_number = k - 1;
+
+  if (binary_array[face_number] == 0) {
+    split = 0; /* the face is not split */ 
+  }
+  else {
+    split = 1; /* the face is split */
+  }
+
+  location[0] = face_number;
+  location[1] = split;
+  location[2] = sub_face_id;
+}
+
+t8_element_shape_t
+t8_default_scheme_sub_c::t8_element_get_shape (const t8_element_t * elem)
+{
+  const t8_quad_with_subelements *pquad_w_sub = (const t8_quad_with_subelements *) elem;
+
+  T8_ASSERT (t8_element_is_valid (elem));
+
+  if (pquad_w_sub->dummy_is_subelement != 1) { 
+    return T8_ECLASS_QUAD;
+  }
+  else { /* for quads, all subelements are triangles */
+    return T8_ECLASS_TRIANGLE;
+  }
+} 
+
+void
+t8_default_scheme_sub_c::t8_element_new (int length, t8_element_t ** elem)
+{
+  /* allocate memory for a quad with subelements */
+  t8_default_scheme_common_c::t8_element_new (length, elem);
+
+  int                 i;
+  for (i = 0; i < length; i++) {
+    t8_quad_with_subelements *pquad_w_sub = (t8_quad_with_subelements *) elem[i];
+    t8_element_init (1, elem[i], 0);
```

Comment:
```suggestion
    t8_element_init (1, elem[i], 1);
```
Last argument should be 1 since `new` was called for the element.

Reply:
Changing this leads to an instant assertion. Need to check again, whats happening here.

Reply:
This problem might be interrelated with the issue below?

Reply:
yes, i think it does not make sense what i said.
Needs to be kept at 0, since the element_new function is still running.